src/tests/ftest/performance/performance_test_base.py

```diff
@@ -0,0 +1,407 @@
+            ["NUM_TARGETS", self.num_targets],
+            ["NUM_CLIENTS", self.num_clients],
+            ["PPC", self.ppn],
+            ["PPN", self.ppn]
+        ]
+
+        # Get group-specific parameters
+        if group == "IOR":
+            params += [
+                ["API", self.ior_cmd.api.value],
+                ["OCLASS", self.ior_cmd.dfs_oclass.value],
+                ["XFER_SIZE", self.ior_cmd.transfer_size.value],
+                ["BLOCK_SIZE", self.ior_cmd.block_size.value],
+                ["SW_TIME", self.ior_cmd.sw_deadline.value],
+                ["CHUNK_SIZE", self.ior_cmd.dfs_chunk.value],
+                ["SEGMENTS", self.ior_cmd.segment_count.value],
+                ["ITERATIONS", self.ior_cmd.repetitions.value]
+            ]
+        elif group == "MDTEST":
+            params += [
+                ["API", self.mdtest_cmd.api.value],
+                ["OCLASS", self.mdtest_cmd.dfs_oclass.value],
+                ["DIR_OCLASS", self.mdtest_cmd.dfs_dir_oclass.value],
+                ["SW_TIME", self.mdtest_cmd.stonewall_timer.value],
+                ["CHUNK_SIZE", self.mdtest_cmd.dfs_chunk.value],
+                ["N_FILE", self.mdtest_cmd.num_of_files_dirs.value],
+                ["BYTES_READ", self.mdtest_cmd.read_bytes.value],
+                ["BYTES_WRITE", self.mdtest_cmd.write_bytes.value],
+                ["MDTEST_FLAGS", self.mdtest_cmd.flags.value]
+            ]
+
+        # Add the extra params
+        if extra_params is not None:
+            params += extra_params
+
+        # Print and align all parameters in the format:
+        # PARAM_NAME : PARAM_VALUE
+        max_len = max([len(param[0]) for param in params])
+        self.log_performance(
+            ["{:<{}} : {}".format(param[0], max_len, param[1]) for param in params]
+        )
+
+    def set_processes_ppn(self, namespace):
+        """Set processes and ppn from a given namespace.
+
+        Args:
+            namespace (str): config namespace. E.g. /run/ior/*
+
+        """
+        self.processes = self.params.get("np", namespace, self.processes)
+        self.ppn = self.params.get("ppn", namespace, self.ppn)
+
+    def verify_system_status(self):
+        """Verify container/pool/system status (in that order)."""
+        if self.pool and self.container:
+            self.log.info(self.daos_cmd.container_query(self.pool.uuid, self.container.uuid))
+        if self.pool:
+            self.display_pool_space()
+        self.dmg_cmd.system_query()
+
+
+    def verify_oclass_server_count(self, oclass):
+        """Verify an object class is compatible with the number of servers.
+
+        Args:
+            oclass (str): The object class. Assumed to be valid.
+
+        """
+        min_servers = oclass_utils.calculate_min_servers(oclass)
+        if self.num_servers < min_servers:
+            self.fail("Need at least {} servers for oclass {}".format(min_servers, oclass))
+
+    def restart_servers(self):
```

Comment:
This is a good candidate for being in a core file, though there is already a restart_servers by a different name. I'd like to somehow combine/rename the two functions as an improvement.